Source/WebKit/UIProcess/WebProcessProxy.messages.in

```diff
@@ -35,6 +35,7 @@ messages -> WebProcessProxy LegacyReceiver {
 
 #if ENABLE(GPU_PROCESS)
     CreateGPUProcessConnection(IPC::ConnectionHandle connectionHandle, struct WebKit::GPUProcessConnectionParameters parameters) AllowedWhenWaitingForSyncReply
+    ResetGPUProcess()
```

Comment:
"Reset" could mean many things. I think we should be clear and call this "TerminateGPUProcess".

Now I'll note that it is fishy to add an IPC that allows the WebContent process to kill the GPUProcess, given that the GPUProcess is shared by all processes. I see that you mitigated that by allowing the WebProcess to only kill the GPUProcess once.

It still feels like a pretty big hack to paper over a real bug somewhere else.

Reply:
It is entirely a big hack to paper over a real bug :) I did mention that in the commit message too.

I don't think we should stop trying to fix the real bug, but this is hopefully a user experience improvement in the interim.